app/assets/javascripts/discourse/app/lib/user-menu/base-item.js

```diff
@@ -26,4 +28,13 @@ export default class UserMenuBaseItem {
   get descriptionClass() {}
 
   get topicId() {}
+
+  onClick({ event, closeUserMenu }) {
+    const href = this.linkHref;
+    if (href) {
+      DiscourseURL.routeTo(href);
+    }
+    closeUserMenu();
```

Comment:
I would probably have this at the top as we are going to do it in any case, rather do it as soon as possible.

Reply:
done!